hnn_core/params.py

```diff
@@ -106,6 +106,177 @@ def read_params(params_fname):
     return params
 
 
+def _long_name(short_name):
+    long_name = dict(L2Basket='L2_basket', L5Basket='L5_basket',
+                     L2Pyr='L2_pyramidal', L5Pyr='L5_pyramidal')
+    if short_name in long_name:
+        return long_name[short_name]
+    return short_name
+
+
+def _short_name(short_name):
+    long_name = dict(L2_basket='L2Basket', L5_basket='L5Basket',
+                     L2_pyramidal='L2Pyr', L5_pyramidal='L5Pyr')
+    if short_name in long_name:
+        return long_name[short_name]
+    return short_name
+
+
+def _extract_bias_specs_from_hnn_params(params, cellname_list):
+    """Create 'bias specification' dicts from saved parameters
+    """
+    bias_specs = {'tonic': {}}  # currently only 'tonic' biases known
+    for cellname in cellname_list:
+        short_name = _short_name(cellname)
+        is_tonic_present = [f'Itonic_{p}_{short_name}_soma' in
+                            params for p in ['A', 't0', 'T']]
+        if any(is_tonic_present):
+            if not all(is_tonic_present):
+                raise ValueError(
+                    f'Tonic input must have the amplitude, '
+                    f'start time and end time specified. One '
+                    f'or more parameter may be missing for '
+                    f'cell type {cellname}')
+            bias_specs['tonic'][cellname] = {
+                'amplitude': params[f'Itonic_A_{short_name}_soma'],
+                't0': params[f'Itonic_t0_{short_name}_soma'],
+                'T': params[f'Itonic_T_{short_name}_soma']
+            }
+    return(bias_specs)
```

Comment:
```suggestion
    return bias_specs
```

just a stylistic nitpick

Reply:
git grep'ed: all of them out now